sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -291,20 +292,16 @@ export class TableClient {
    * ```
    */
   // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
-  public async deleteTable(options: OperationOptions = {}): Promise<void> {
-    const { span, updatedOptions } = createSpan("TableClient-deleteTable", options);
-    try {
-      await this.table.delete(this.tableName, updatedOptions);
-    } catch (e) {
-      if (e.statusCode === 404) {
-        logger.info("TableClient-deleteTable: Table doesn't exist");
-      } else {
-        span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
-        throw e;
```

Comment:
😓 we missed re-throwing if the status code is not 404. I'll send out a PR to fix this soon